logback-core/src/main/java/ch/qos/logback/core/net/HardenedObjectInputStream.java

```diff
@@ -20,20 +20,22 @@
  */
 public class HardenedObjectInputStream extends ObjectInputStream {
 
-    final List<String> whitelistedClassNames;
-    final static String[] JAVA_PACKAGES = new String[] { "java.lang", "java.util" };
+    final private List<String> whitelistedClassNames;
+    final private static String[] JAVA_PACKAGES = new String[] { "java.lang", "java.util" };
+    final private static int DEPTH_LIMIT = 16;
+    final private static int ARRAY_LIMIT = 10000;
```

Comment:
you forgot to use those. Did you have a merge conflict when cherry-picking 9c782b45be4abdafb7e17481e24e7354c2acd1eb and 2cd8cabc5b1289d8efab7cb0f728010c8f381e71 ?

Reply:
> you forgot to use those. Did you have a merge conflict when cherry-picking https://github.com/qos-ch/logback/commit/9c782b45be4abdafb7e17481e24e7354c2acd1eb and https://github.com/qos-ch/logback/commit/2cd8cabc5b1289d8efab7cb0f728010c8f381e71 ?

Just for the record, although the PR has been already closed.

No I had no merge conflict for them. The fix of this CVE has been _first_ announced for `1.13.12`:

https://logback.qos.ch/news.html#1.3.12

So that I only cherry-picked the green commits below (from `branch_1.3.x`), but apparently there was more to it (the red commit by the `1.13.14` release) which you're asking about.

But apparently the fix had _even much more to it_ than just cherry-picking as we see in https://github.com/qos-ch/logback/commit/bb095154be011267b64e37a1d401546e7cc2b7c3.

<img width="577" alt="Screenshot 2023-12-04 at 08 48 54" src="https://github.com/qos-ch/logback/assets/3122177/c2b7bec5-44f2-4f49-ad5c-43fbf747e2de">